src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useCallback, useContext, useMemo, useState, useRef } from "react";
+
+import type {
+  ViewerContextType,
+  ViewerState,
+  ViewerSettingChangeHandlers,
+  ViewerSettingUpdater,
+  ChannelSettingUpdater,
+  ChannelState,
+  MultipleChannelSettingsUpdater,
+  PartialIfObject,
+} from "./types";
+import { ImageType, RenderMode, ViewMode } from "../../shared/enums";
+import {
+  ALPHA_MASK_SLIDER_DEFAULT,
+  BACKGROUND_COLOR_DEFAULT,
+  BOUNDING_BOX_COLOR_DEFAULT,
+  BRIGHTNESS_SLIDER_LEVEL_DEFAULT,
+  DENSITY_SLIDER_LEVEL_DEFAULT,
+  INTERPOLATION_ENABLED_DEFAULT,
+  LEVELS_SLIDER_DEFAULT,
+} from "../../shared/constants";
+import { ColorArray } from "../../shared/utils/colorRepresentations";
+
+const isObject = <T,>(val: T): val is Extract<T, Record<string, unknown>> =>
+  typeof val === "object" && val !== null && !Array.isArray(val);
```

Comment:
are we really still needing to write our own isObject function in 2024?

Reply:
JavaScript 🙃